src/ContentCreator.php

```diff
@@ -547,38 +602,37 @@ protected function getBundleReferencedFieldDefinitions($entity_type, $bundle_typ
   protected function createEntityReferenceFields($entity_type, $bundle_type, $start_entity_id, $end_entity_id) {
     $referenced_field_definitions = $this->getBundleReferencedFieldDefinitions($entity_type, $bundle_type);
     foreach ($referenced_field_definitions as $referenced_field_definition) {
-      $target_entity_type = $referenced_field_definition['target_type'];
       $histogram = $referenced_field_definition['histogram'];
+      if (empty($histogram)) {
+        continue;
+      }
+
+      $target_entity_type = $referenced_field_definition['target_type'];
       $reference_type = $referenced_field_definition['reference_type'];
       $table_name = $referenced_field_definition['table_name'];
       $rev_table_name = $referenced_field_definition['rev_table_name'];
       $field_name = $referenced_field_definition['field_name'];
       for ($entity_id = $start_entity_id; $entity_id < $end_entity_id; $entity_id++) {
-        $target_entity_bundle = $this->getRandomWeighted($this->getEntityBundleInstances($target_entity_type));
-
-        // Block creation early.
-        if (isset($this->entityTypeReferenceNestingStack[$target_entity_type . '_' . $target_entity_bundle])) {
-          continue;
-        }
-
         $num_of_instances = $this->getRandomWeighted($histogram);
 
-        // In case of recursive nesting, we are not going to add entities.
-        if ($this->createBundle($target_entity_type, $target_entity_bundle, $num_of_instances, $entity_id, $entity_type, $field_name) === 0) {
-          continue;
+        $target_entity_bundles = [];
+        for ($instance_index = $num_of_instances; $instance_index >= 0; $instance_index--) {
+          $target_entity_bundles[] = $this->getRandomWeighted($this->getEntityBundleInstancesForField($entity_type, $bundle_type, $field_name));
```

Comment:
`getEntityBundleInstancesForField` can return empty array, that will create empty values for `$target_entity_bundles`. If such an `$target_entity_bundles` array is created, the foreach below will have empty `$target_entity_bundle` and `$this->createBundleForReferenceField` will throw "Undefined index:" errors.
I have created a report, which leads to that error. I will send it to you, so you can investigate this.

Reply:
The conclusion from the chat:
- in the content creator, we will skip the creation of reference if target bundles are empty
- in distribution, we will add `media_entity_generic` in built artifact
- in Thunder performance docker we will enable `media_entity_generic`

Reply:
- the skip of reference creation is added to this PR
- PR for distribution is provided here: https://github.com/thunder/thunder-distribution/pull/67
- and enabling of module for Thunder performance docker is added in this PR: https://github.com/thunder/docker-thunder-performance/pull/2